src/Sylius/Bundle/ApiBundle/CommandHandler/RegisterShopUserHandler.php

```diff
@@ -58,6 +66,18 @@ public function __invoke(RegisterShopUser $command): void
         $customer->setPhoneNumber($command->phoneNumber);
         $customer->setUser($user);
 
+        $this->handleVerification($user);
+
         $this->shopUserManager->persist($user);
     }
+
+    private function handleVerification(ShopUserInterface $user): void
+    {
+        /** @var ChannelInterface $channel */
+        $channel = $this->channelContext->getChannel();
```

Comment:
In the long term, I would prefer to skip the usage of contexts in handlers. I think we should put data like channel into command.

Reply:
Hmm, do you have any resolution for this that we can implement now? Creating this function i was basing on ShopBundle->UserRegistrationListener. Should we create something similar or we leave it as it is and go back to it, when we make API user verification feauture? @lchrusciel